airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GcsDestination.java

```diff
@@ -90,4 +98,16 @@ public AirbyteMessageConsumer getConsumer(final JsonNode config,
         configuredCatalog);
   }
 
+  /**
+   * The method is used to verify that user set location that match to actual bucket's location if it
+   * already exists
+   */
+  private void assertRegion(final String regionFromConfig, final String actualBucketRegion) {
+    if (!regionFromConfig.equals(actualBucketRegion)
+        && !regionFromConfig.contains(actualBucketRegion) // this check is needed to pass the case like "US" vs "US-WEST-1"
+        && !actualBucketRegion.contains(regionFromConfig)) {
+      throw new ConfigErrorException(MISMATCH_LOCATIONS_EXCEPTION_MESSAGE);
```

Comment:
Could be nice to include the mismatched values in the error message

Reply:
Updated, thanks